src/uucore/src/lib/features/fs.rs

```diff
@@ -141,13 +141,13 @@ fn resolve<P: AsRef<Path>>(original: P) -> IOResult<PathBuf> {
 /// a path.
 /// The `miss_mode` parameter controls how missing path elements are handled
 ///
-/// * [`MissingHandling::Normal`] makes this function behave like
+/// * [`MissingHandling::Strict`] makes this function behave like
 ///   [`std::fs::canonicalize`], resolving symbolic links and returning
 ///   an error if the path does not exist.
-/// * [`MissingHandling::Missing`] makes this function ignore non-final
+/// * [`MissingHandling::Allowed`] makes this function ignore non-final
 ///   components of the path that could not be resolved.
-/// * [`MissingHandling::Existing`] makes this function return an error
-///   if the final component of the path does not exist.
+/// * [`MissingHandling::Final`] makes this function ignore final components
```

Comment:
Wait... doesn't it do the exact opposite? If `miss_mode` is `MissingHandling::Final` it does _not_ ignore the error (line 233). `MissingHandling::Strict`, on the other hand ignores the error for the last component.

To summarize, there are three `MissingHandling` options:
- allow no component that could not be resolved (`MissingHandling::Final`)
- allow the last component to not be resolvable (`MissingHandling::Strict`)
- allow all components to not be resolvable (`MissingHandling::Allowed`)

I'm not sure these names make a lot of sense, maybe you could look at this again?

Reply:
I had the code the wrong way round which is fixed now, these descriptions are correct. The intended system is:
* `Strict` everything must exist
* `Final` last part can be missing
* `Allowed` anything can be missing